src/opendr/perception/face_recognition/face_recognition_learner.py

```diff
@@ -455,7 +455,7 @@ def infer(self, img):
         """
         if not isinstance(img, Image):
             img = Image(img)
```

Comment:
This is a bit dangerous. I would rather raise a TypeError.

Reply:
Indeed. However, this is already extensively used in other implementations to allow for handling OpenCV arrays without adding boilerplate code. I will add an extra check to ensure that data can be cast into images directly into the constructor in order to raise the error from there.